ebms-provider/src/main/kotlin/no/nav/emottak/ebms/Routes.kt

```diff
@@ -118,16 +132,61 @@ fun Route.postEbmsSync(
                 it.signer(signatureDetails)
             }
             log.info(ebmsMessage.marker(), "Created MessageError response")
-            // TODO: Event-logging Feil
+
+            eventLoggingService.registerEvent(
+                EventType.ERROR_WHILE_SENDING_MESSAGE_VIA_HTTP,
+                ebMSDocument,
+                mapOf("error_message" to (ebmsException.message ?: "Unknown error"))
+            )
+
             call.respondEbmsDokument(it)
             return@post
         }
     } catch (ex: Exception) {
         log.error(ebmsMessage.marker(), "Unknown error during message processing: ${ex.message}", ex)
-        // TODO: Event-logging Feil
+
+        eventLoggingService.registerEvent(
+            EventType.ERROR_WHILE_SENDING_MESSAGE_VIA_HTTP,
+            ebMSDocument,
+            mapOf("error_message" to (ex.message ?: "Unknown error"))
+        )
+
         call.respond(
             HttpStatusCode.InternalServerError,
             ex.parseAsSoapFault()
         )
     }
 }
+
+@OptIn(ExperimentalUuidApi::class)
+suspend fun EventLoggingService.registerEvent(
+    eventType: EventType,
+    ebMSDocument: EbMSDocument,
+    eventData: Map<String, String> = emptyMap()
+) {
+    log.debug("Event reg. test: Registering event for requestId: ${ebMSDocument.requestId}")
+
+    try {
+        val requestId = try {
+            Uuid.parse(ebMSDocument.requestId)
+        } catch (e: Exception) {
+            Uuid.random()
+        }
```

Comment:
Enda et forslag (beklager at jeg kommer med så mange tanker i dag 😅):
Jeg tror vi har brukt lignende logikk flere steder for å sikre at vi alltid får en gyldig UUID. Kunne det vært nyttig å lage en extension-metode for dette også? For eksempel noe som:

`val requestId = Uuid.safeParseOrGenerate(ebMSDocument.requestId)`

Den kunne da håndtert try-catch-logikken og returnert en gyldig UUID enten ved å parse verdien, eller ved å generere en ny. Da blir koden vår litt ryddigere også :crossed_fingers: